scrapy/dispatch/weakref_backports.py

```diff
@@ -0,0 +1,67 @@
+"""
+weakref_backports is a partial backport of the weakref module for python
+versions below 3.4.
```

Comment:
I glanced through the code in its repo here: https://github.com/pjdelport/backports.weakref/blob/master/src/backports/weakref.py and it looks like he has implemented the backport himself. Ours is a verbatim copy of the code in the Python 3.4 source. I can do a build with that module but it's your call in the end on which implementation of the backport you'd like to go with.